clang/test/dpct/thrust-for-h2o4gpu-specific-case.cu

```diff
@@ -11,24 +11,29 @@
 // CHECK-NEXT: #include <dpct/dpl_utils.hpp>
 #include <thrust/copy.h>
 #include <thrust/device_vector.h>
+#include <thrust/device_allocator.h>
+#include <thrust/device_malloc_allocator.h>
+#include <thrust/iterator/iterator_traits.h>
 
-//CHECK: template <typename T>
-//CHECK-NEXT:void foo_cpy(dpct::device_vector<T, sycl::buffer_allocator<T>> &Do, dpct::device_vector<T, sycl::buffer_allocator<T>> &Di) {
+//CHECK:template <typename T>
+//CHECK-NEXT:void foo_cpy(dpct::device_vector<T, dpct::deprecated::usm_device_allocator<T>> &Do, dpct::device_vector<T, dpct::deprecated::usm_device_allocator<T>> &Di) {
 //CHECK-NEXT: return;
 //CHECK-NEXT:}
 template <typename T>
 void foo_cpy(thrust::device_vector<T, thrust::device_malloc_allocator<T>> &Do, thrust::device_vector<T, thrust::device_malloc_allocator<T>> &Di) {
  return;
 }
 
-//CHECK: void foo_1() {
-//CHECK-NEXT: dpct::device_vector<int, sycl::buffer_allocator<int>> **tt=NULL;
-//CHECK-NEXT: dpct::device_vector<int, sycl::buffer_allocator<int>> *dd[10];
-//CHECK-NEXT: foo_cpy(*tt[0], *dd[0]);
+//CHECK:void foo_1() {
+//CHECK-NEXT:  dpct::device_vector<int, dpct::deprecated::usm_device_allocator<int>> **tt=NULL;
+//CHECK-NEXT:  dpct::device_vector<int, dpct::deprecated::usm_device_allocator<int>> *dd[10];
+//CHECK-NEXT:  foo_cpy(*tt[0], *dd[0]);
 //CHECK-NEXT:}
 void foo_1() {
   thrust::device_vector<int, thrust::device_malloc_allocator<int>> **tt=NULL;
   thrust::device_vector<int, thrust::device_malloc_allocator<int>> *dd[10];
   foo_cpy(*tt[0], *dd[0]);
 }
 
+//CHECK:namespace foo_struct { template <typename T, typename MemorySpace> struct default_memory_allocator : dpct::deprecated::usm_device_allocator<T>{};}
+namespace foo_struct { template <typename T, typename MemorySpace> struct default_memory_allocator : thrust::device_malloc_allocator<T>{};}
```

Comment:
is the mapping right:  using shared alloctor to migrate thrust::device_malloc_allocator?

Reply:
Current map type is not exactly semantically equivalent， will use new implantation in the update PR.